README.md

```diff
@@ -1,2 +1,128 @@
+
+### Set up an IDE (optional)
+
+If you want an IDE deployed (e.g. working on a server). Install [a vscode service](https://artifacthub.io/packages/helm/inseefrlab/vscode) via Helm, then access by port-forwarding `kubectl -n argo port-forward deployment/my-vscode 35547:8080`. You will probably have to give admin rights to this VSCode (service account: `my-vscode` over MinIO (as done for Argo)). Finally, you will also have to add Metaflow configuration variables to the ConfigMap of VSCode, for putting metaflow code packages in MinIO. 
+
+![configmap-my-vscode-configmaps3metaflow](https://github.com/SDSC-ORD/zarr-linked-data/assets/22447169/498a1306-5289-4527-9cfa-22e947d555e3)
+
+Then you just install the requirements with `pip install -r requirements.txt` and you're set to go! 
+
+# C. Example/Usage
+
+## Fake Zarr Data
+You may not have data under the zarr format yet. You can define some metadata instance objects (such as in `zarr_linked_data/data/original_data.jsonld`), then define your hierarchy levels, data level and other parameters for the creation of a random Zarr test store using `zarr_linked_data/fake_data_flow.py`.
+You can find all our example data in the `data` folder, from the original jsonld metadata instance data, to the `test_store.zarr` containing random arrays as well as the jsonld metadata.
+
+## Local scripts
+
+- Fake data creation pipeline with: `python zarr_linked_data/fake_data_flow.py`
+- Metadata consolidation Metaflow pipeline with: `python zarr_linked_data/local_dev/metadata_consolidate_metaflow.py run`
+- Retrieve URI Metaflow pipeline with: `python zarr_linked_data/local_dev/uri_matching_metaflow.py run`
+
+## Local runthrough step-by-step 
+
+Here is a detailed run through using poetry to set-up dependencies. (Please first install poetry and run `poetry install` as explained in `B. Getting Started`).
+
+1. Start your environment with `poetry shell`
+
+2. Run: `poetry run python zarr_linked_data/fake_data_flow.py`
+   Goal: You don't have any data? No problem, run this script to generate a `test_store.zarr` (You can personalize the script to make it look like the data you expect to handle.)
+  
+3. Run: `poetry run python zarr_linked_Data/local_dev/metadata_consolidate_metaflow.py run --path_for_store="zarr_linked_data/data/test_store.zarr"`
+   Goal: Create the Zarr metadata store `.all_metadata` for the Zarr test_store i.e. a JSON file containing the metadata for the entire store.
+  
+4. Run: `poetry run python zarr_linked_data/local_dev/uri_matching_metaflow.py run --path_for_store="zarr_linked_data/data/test_store2.zarr" --uri "http://www.catplus.ch/ontology/concepts/sample1" --path_save="zarr_linked_data/data/results/dataset.npy"`
```

Comment:
```suggestion
4. Run: `poetry run python zarr_linked_data/local_dev/uri_matching_metaflow.py run --path_for_store="zarr_linked_data/data/test_store.zarr" --uri "http://www.catplus.ch/ontology/concepts/sample1" --path_save="zarr_linked_data/data/results/dataset.npy"`
```

**fix:** I think this is the wrong path, shouldn't we reuse the consolidated data form the previous step